cardsavr_e2e/CardSavrHelperTests.cs

```diff
@@ -0,0 +1,104 @@
+            );
+            await helper.CloseSession(config.app_username);
+            Assert.NotNull(login.cardholder.grant);
+            Assert.Equal(login.cardholder.cuid, cuid);
+            Assert.True(login.card.id > 0);
+            log.Info("CARDSAVRHELPERTESTS cuid: " + login.cardholder.cuid + ", grant: " + login.grant + ", card_id: " + login.card.id);
+            //login can now be used as a redirect to a url that can log in and process jobs
+
+        }
+
+        private static readonly log4net.ILog log = log4net.LogManager.GetLogger(
+            System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+
+    }
+}
+
+
+/*
+using System;
+using System.Collections.Generic;
+using System.Collections.Specialized;
+using System.Threading.Tasks;
+using System.Net.Http;
+using System.Dynamic;
+
+using Switch.CardSavr.Http;
+using Newtonsoft.Json;
+
+namespace cardsavr_e2e
+{
+    public class CardSavrHelperOps : OperationBase
+    {
+        public CardSavrHelperOps()
+        {
+        }
+
+        protected static readonly log4net.ILog log = log4net.LogManager.GetLogger(
+            System.Reflection.MethodBase.GetCurrentMethod().DeclaringType);
+
+        public override async Task Execute(CardSavrHttpClient http, Context ctx, params object[] extra)
+        {
+            CardsavrHelper helper = new CardsavrHelper();
+            helper.SetAppSettings(CardsavrSession.accountBaseUrl, Context.accountCardholderAgentAppID, Context.accountCardholderAgentStaticKey, CardsavrSession.rejectUnauthorized);
+            await helper.LoginAndCreateSession(CardsavrSession.accountCardholderAgentUserName, Context.accountCardholderAgentPassword, "{\"key\": \"my_trace\"}");
+            
+            PropertyBag cd = new PropertyBag(){{"my_fi", new PropertyBag(){{"token", "123"}}}};
+
+            ClientLogin login = await helper.CreateCard(CardsavrSession.accountCardholderAgentUserName, "default", 
+                new User(){ email = "foo@foo.com", phone_number = "5555555555", custom_data = cd, username = $"{CardsavrSession.e2e_identifier}_{CardsavrSession.random.Next(100)}_0" },
+                new Card(){ first_name="Strivve", last_name="User", pan="4111111111111111", cvv="111", expiration_month="01", expiration_year="25" },
+                new Address(){ is_primary=true, address1="1234 1st ave", city="Seattle", subnational="WA", postal_code="98006", country="USA" }
+            );
+            await helper.CloseSession(CardsavrSession.accountCardholderAgentUserName);
+            log.Info("CARDSAVRHELPEROPS username: " + login.cardholder.username + ", grant: " + login.userCredentialGrant + ", card_id: " + login.card.id);
+            //login can now be used as a redirect to a url that can log in and process jobs
+        }
+
+        public override async Task Cleanup(CardSavrHttpClient http, Context ctx)
+        {
+            await base.Cleanup(http, ctx);
+        }
+
+    }
+}*/
```

Comment:
What's this?

Reply:
It's basically exactly what Advancial does.  (In fact, it's all that they do)

Reply:
But now I'm looking, the class should be renamed.  The way the framework is, the filename is irrelevant.

Reply:
Oh I see, you're talking about the commented out code.  That's the old test, I just removed it.